From f7767014b984fae5df9b9c0d6b0df5fe10aba360 Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Tue, 16 Aug 2011 14:03:17 +0200
Subject: [PATCH] Make sure we pass ENABLE_EDS to vala flags

---
 folks/Makefile.am |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/folks/Makefile.am b/folks/Makefile.am
index 1fde141..a4d03ea 100644
--- a/folks/Makefile.am
+++ b/folks/Makefile.am
@@ -44,6 +44,10 @@ libfolks_la_SOURCES = \
 	object-cache.vala \
 	$(NULL)
 
+if ENABLE_EDS
+EDS_FLAGS = -D ENABLE_EDS
+endif
+
 libfolks_la_VALAFLAGS = \
 	--gir=Folks-$(API_VERSION).gir \
 	--library=folks \
@@ -57,6 +61,7 @@ libfolks_la_VALAFLAGS = \
 	--includedir folks \
 	--vapi folks.vapi \
 	-H folks.h \
+	$(EDS_FLAGS) \
 	$(NULL)
 
 libfolks_la_CFLAGS = \
-- 
1.7.6

