diff -urp folks-0.2.1.OLD/tests/telepathy/Makefile.am folks-0.2.1/tests/telepathy/Makefile.am
--- folks-0.2.1.OLD/tests/telepathy/Makefile.am	2010-10-25 17:58:04.000000000 -0400
+++ folks-0.2.1/tests/telepathy/Makefile.am	2010-10-29 18:27:56.570518002 -0400
@@ -46,6 +46,7 @@ contact_retrieval_SOURCES = \
 	$(NULL)
 
 contact_retrieval_LDADD = \
+	$(GEE_LIBS) \
 	$(top_builddir)/tests/lib/telepathy/contactlist/libtp-test-contactlist.la \
 	$(top_builddir)/folks/libfolks.la
 
diff -urp folks-0.2.1.OLD/tests/telepathy/Makefile.in folks-0.2.1/tests/telepathy/Makefile.in
--- folks-0.2.1.OLD/tests/telepathy/Makefile.in	2010-10-25 17:58:26.000000000 -0400
+++ folks-0.2.1/tests/telepathy/Makefile.in	2010-10-29 18:26:45.634518001 -0400
@@ -282,6 +282,7 @@ contact_retrieval_SOURCES = \
 	$(NULL)
 
 contact_retrieval_LDADD = \
+	$(GEE_LIBS) \
 	$(top_builddir)/tests/lib/telepathy/contactlist/libtp-test-contactlist.la \
 	$(top_builddir)/folks/libfolks.la
 
