diff -up folks-0.12.1/backends/eds/lib/edsf-persona-store.vala.eds-libebook-api folks-0.12.1/backends/eds/lib/edsf-persona-store.vala
--- folks-0.12.1/backends/eds/lib/edsf-persona-store.vala.eds-libebook-api	2019-04-24 13:20:07.000000000 +0200
+++ folks-0.12.1/backends/eds/lib/edsf-persona-store.vala	2019-05-21 15:13:30.384790173 +0200
@@ -610,7 +610,7 @@ public class Edsf.PersonaStore : Folks.P
           /* _addressbook is guaranteed to be non-null before we ensure that
            * prepare() has already been called. */
           yield ((!) this._addressbook).remove_contact (
-              ((Edsf.Persona) persona).contact, null);
+              ((Edsf.Persona) persona).contact, E.BookOperationFlags.NONE, null);
         }
       catch (GLib.Error e)
         {
@@ -1258,7 +1258,7 @@ public class Edsf.PersonaStore : Folks.P
 
           /* Commit the new contact. _addressbook is asserted as being non-null
            * above. */
-          yield ((!) this._addressbook).add_contact (contact, null,
+          yield ((!) this._addressbook).add_contact (contact, E.BookOperationFlags.NONE, null,
               out added_uid);
 
           debug ("Finished sending new contact to EDS; received UID %s.",
@@ -1377,7 +1377,7 @@ public class Edsf.PersonaStore : Folks.P
 
           /* Commit the modification. _addressbook is asserted as being non-null
            * above. */
-          yield ((!) this._addressbook).modify_contact (contact, null);
+          yield ((!) this._addressbook).modify_contact (contact, E.BookOperationFlags.NONE, null);
 
           timeout_id = Timeout.add_seconds (PersonaStore._property_change_timeout, () =>
             {
diff -up folks-0.12.1/meson.build.eds-libebook-api folks-0.12.1/meson.build
--- folks-0.12.1/meson.build.eds-libebook-api	2019-04-24 13:20:07.000000000 +0200
+++ folks-0.12.1/meson.build	2019-05-21 15:13:30.384790173 +0200
@@ -67,7 +67,7 @@ inspect_tool_enabled = get_option('inspe
 # Dependencies
 #-------------------------------------------------
 min_glib_version = '2.44'
-min_eds_version = '3.13.90'
+min_eds_version = '3.33.2'
 
 valac = meson.get_compiler('vala')
 cc = meson.get_compiler('c')
diff -up folks-0.12.1/tests/eds/helper-create-many-contacts.vala.eds-libebook-api folks-0.12.1/tests/eds/helper-create-many-contacts.vala
--- folks-0.12.1/tests/eds/helper-create-many-contacts.vala.eds-libebook-api	2019-04-24 13:20:07.000000000 +0200
+++ folks-0.12.1/tests/eds/helper-create-many-contacts.vala	2019-05-21 15:13:30.384790173 +0200
@@ -107,7 +107,7 @@ public class Main
       SList<string> uids;
       try
         {
-          book_client.add_contacts_sync (contacts, out uids, null);
+          book_client.add_contacts_sync (contacts, E.BookOperationFlags.NONE, out uids, null);
         }
       catch (Error e)
         {
diff -up folks-0.12.1/tests/eds/helper-delete-contacts.vala.eds-libebook-api folks-0.12.1/tests/eds/helper-delete-contacts.vala
--- folks-0.12.1/tests/eds/helper-delete-contacts.vala.eds-libebook-api	2019-04-24 13:20:07.000000000 +0200
+++ folks-0.12.1/tests/eds/helper-delete-contacts.vala	2019-05-21 15:13:30.384790173 +0200
@@ -41,7 +41,7 @@ public class Main
       SList<string> uids;
       book_client.get_contacts_uids_sync (
           "(contains \"x-evolution-any-field\" \"\")", out uids);
-      book_client.remove_contacts_sync (uids);
+      book_client.remove_contacts_sync (uids, E.BookOperationFlags.NONE);
     }
 
   private static string _uid = "";
diff -up folks-0.12.1/tests/lib/eds/backend.vala.eds-libebook-api folks-0.12.1/tests/lib/eds/backend.vala
--- folks-0.12.1/tests/lib/eds/backend.vala.eds-libebook-api	2019-04-24 13:20:07.000000000 +0200
+++ folks-0.12.1/tests/lib/eds/backend.vala	2019-05-21 15:13:30.384790173 +0200
@@ -71,7 +71,7 @@ public class EdsTest.Backend
         {
           yield this._addressbook.get_contact (uid, null, out contact);
           this._set_contact_fields (contact, updated_data);
-          yield this._addressbook.modify_contact (contact, null);
+          yield this._addressbook.modify_contact (contact, E.BookOperationFlags.NONE, null);
         }
       catch (GLib.Error e)
         {
@@ -86,7 +86,7 @@ public class EdsTest.Backend
       try
         {
           yield this._addressbook.get_contact (uid, null, out contact);
-          yield this._addressbook.remove_contact (contact, null);
+          yield this._addressbook.remove_contact (contact, E.BookOperationFlags.NONE, null);
         }
       catch (GLib.Error e)
         {
@@ -211,7 +211,7 @@ public class EdsTest.Backend
         {
           GLib.SList<string> uids;
 
-          yield this._addressbook.add_contacts (contacts, null, out uids);
+          yield this._addressbook.add_contacts (contacts, E.BookOperationFlags.NONE, null, out uids);
 
           foreach (unowned string uid in uids)
             this._e_contacts += uid;
diff -up folks-0.12.1/tests/lib/eds/test-case.vala.eds-libebook-api folks-0.12.1/tests/lib/eds/test-case.vala
--- folks-0.12.1/tests/lib/eds/test-case.vala.eds-libebook-api	2019-05-21 15:13:42.605790004 +0200
+++ folks-0.12.1/tests/lib/eds/test-case.vala	2019-05-21 15:14:03.266789718 +0200
@@ -123,6 +123,7 @@ public class EdsTest.TestCase : Folks.Te
         };
       const string address_book_services[] =
         {
+          "org.gnome.evolution.dataserver.AddressBook10",
           "org.gnome.evolution.dataserver.AddressBook9",
           "org.gnome.evolution.dataserver.AddressBook8",
           "org.gnome.evolution.dataserver.AddressBook7",
