diff -up folks-0.6.2.1/folks/Makefile.am.typelib folks-0.6.2.1/folks/Makefile.am
--- folks-0.6.2.1/folks/Makefile.am.typelib	2011-09-14 17:16:03.150226985 -0400
+++ folks-0.6.2.1/folks/Makefile.am	2011-09-14 17:16:26.251435649 -0400
@@ -118,7 +118,7 @@ GITIGNOREFILES = \
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
 INTROSPECTION_SCANNER_ARGS =
-INTROSPECTION_COMPILER_ARGS = -l libfolks.so
+INTROSPECTION_COMPILER_ARGS = -l libfolks.so.25
 
 if HAVE_INTROSPECTION
 Folks-$(API_VERSION).gir: libfolks.la
