diff -urp folks-0.6.8.OLD/backends/telepathy/lib/tpf-persona.vala folks-0.6.8/backends/telepathy/lib/tpf-persona.vala
--- folks-0.6.8.OLD/backends/telepathy/lib/tpf-persona.vala	2012-03-26 18:20:57.000000000 -0400
+++ folks-0.6.8/backends/telepathy/lib/tpf-persona.vala	2012-03-30 09:37:50.425562828 -0400
@@ -614,7 +614,7 @@ public class Tpf.Persona : Folks.Persona
     {
       unowned string id = contact.get_identifier ();
       var connection = contact.connection;
-      var account = this._account_for_connection (connection);
+      var account = connection.get_account ();
       var uid = this.build_uid (store.type_id, store.id, id);
 
       Object (contact: contact,
@@ -663,7 +663,7 @@ public class Tpf.Persona : Folks.Persona
 
       /* Set our single IM address */
       var connection = this.contact.connection;
-      var account = this._account_for_connection (connection);
+      var account = connection.get_account ();
 
       try
         {
@@ -1005,25 +1005,6 @@ public class Tpf.Persona : Folks.Persona
         }
     }
 
-  private static Account? _account_for_connection (Connection conn)
-    {
-      var manager = AccountManager.dup ();
-      var accounts = manager.get_valid_accounts ();
-
-      Account account_found = null;
-      accounts.foreach ((l) =>
-        {
-          unowned Account account = (Account) l;
-          if (account.connection == conn)
-            {
-              account_found = account;
-              return;
-            }
-        });
-
-      return account_found;
-    }
-
   private void _contact_notify_presence_message ()
     {
       this.presence_message = this.contact.get_presence_message ();
diff -urp folks-0.6.8.OLD/configure.ac folks-0.6.8/configure.ac
--- folks-0.6.8.OLD/configure.ac	2012-03-26 18:20:57.000000000 -0400
+++ folks-0.6.8/configure.ac	2012-03-30 09:37:50.433562372 -0400
@@ -145,7 +145,7 @@ AM_CONDITIONAL([ENABLE_LIBSOCIALWEB],
 # -----------------------------------------------------------
 
 GLIB_REQUIRED=2.24.0
-TP_GLIB_REQUIRED=0.13.1
+TP_GLIB_REQUIRED=0.15.5
 VALA_REQUIRED=0.15.2
 VALADOC_REQUIRED=0.3.1
 TRACKER_SPARQL_MAJOR=0.14
diff -urp folks-0.6.8.OLD/NEWS folks-0.6.8/NEWS
--- folks-0.6.8.OLD/NEWS	2012-03-26 18:20:57.000000000 -0400
+++ folks-0.6.8/NEWS	2012-03-30 09:38:44.679470930 -0400
@@ -11,6 +11,7 @@ Major changes:
 * Improvements to folks-inspect to make it a good command line citizen
 
 Bugs fixed:
+* Bug 672980 — tp_account_get_protocol: assertion `TP_IS_ACCOUNT (account)' failed
 * Bug 670191 — IndividualAggregator::is-quiescent is never set to TRUE
 * Bug 670523 — persona.vala:153:folks_persona_get_individual:
   assertion failed: (_tmp6_)
