diff -up folks-0.6.3/folks/Makefile.am.typelib folks-0.6.3/folks/Makefile.am
--- folks-0.6.3/folks/Makefile.am.typelib	2011-09-21 20:51:43.708606471 -0400
+++ folks-0.6.3/folks/Makefile.am	2011-09-21 20:52:08.668607902 -0400
@@ -121,7 +121,7 @@ MAINTAINERCLEANFILES =
 # We have to extract our own dlname from libfolks.la; see bgo#658002.
 # This is what g-ir-scanner does.
 libfolks_dlname = \
-	`$(SED) -n "s/^dlname='\([A-Za-z0-9.-+]\+\)'/\1/p" libfolks.la`
+	`$(SED) -n "s/^dlname='\([A-Za-z0-9.+-]\+\)'/\1/p" libfolks.la`
 
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
diff -up folks-0.6.3/folks/Makefile.in.typelib folks-0.6.3/folks/Makefile.in
--- folks-0.6.3/folks/Makefile.in.typelib	2011-09-21 20:52:32.831609310 -0400
+++ folks-0.6.3/folks/Makefile.in	2011-09-21 20:52:42.554609839 -0400
@@ -465,7 +465,7 @@ MAINTAINERCLEANFILES = $(am__append_1) $
 # We have to extract our own dlname from libfolks.la; see bgo#658002.
 # This is what g-ir-scanner does.
 libfolks_dlname = \
-	`$(SED) -n "s/^dlname='\([A-Za-z0-9.-+]\+\)'/\1/p" libfolks.la`
+	`$(SED) -n "s/^dlname='\([A-Za-z0-9.+-]\+\)'/\1/p" libfolks.la`
 
 INTROSPECTION_GIRS = 
 INTROSPECTION_SCANNER_ARGS = 
